homeassistant/components/modbus/switch.py

```diff
@@ -62,14 +59,9 @@ def __init__(self, hub: ModbusHub, config: dict):
         self._available = True
         self._scan_interval = timedelta(seconds=config[CONF_SCAN_INTERVAL])
         self._address = config[CONF_ADDRESS]
-        if config[CONF_WRITE_TYPE] == CALL_TYPE_COIL:
-            self._async_write_func = self._hub.async_write_coil
-            self._command_on = 0x01
```

Comment:
Where do we handle this case and setting now?

Reply:
the part command_on/command_off are set as default in the schema (see line 63-64) so these lines are not needed.

The self._hub.async_write_coil and async_write_register are replaced by self._hub.async_pyhub_call(... self._write_type) line 111-112 and 127-128. So need for self._async_write_func.

Similar pattern for the read part.

In short a lot simpler code.